src/Engine/Graphics/Indoor.cpp

```diff
@@ -2263,37 +2259,64 @@ void BLV_ProcessPartyActions() {  // could this be combined with odm process act
         pParty->uFallSpeed = fixpoint_mul(58500, pParty->uFallSpeed);
     }
 
-    //  //Воспроизведение звуков ходьбы/бега-------------------------
-    uint pX_ = abs(pParty->vPosition.x - new_party_x);
-    uint pY_ = abs(pParty->vPosition.y - new_party_y);
-    uint pZ_ = abs(pParty->vPosition.z - new_party_z);
-    if (engine->config->settings.WalkSound.value() && pParty->walk_sound_timer <= 0) {
-        pAudioPlayer->StopAll(804);  // stop sound
-        if (party_running_flag && (!hovering || not_high_fall)) {  // Бег и (не прыжок или не высокое падение )
-            if (integer_sqrt(pX_ * pX_ + pY_ * pY_ + pZ_ * pZ_) >= 16) {
-                if (on_water)
-                    pAudioPlayer->playWalkSound(SOUND_RunWaterIndoor);
-                else if (pIndoor->pFaces[uFaceID].uAttributes & FACE_INDOOR_CARPET)  //по ковру
-                    pAudioPlayer->playWalkSound(SOUND_RunCarpet);
-                else
-                    pAudioPlayer->playWalkSound(SOUND_RunWood);
-                pParty->walk_sound_timer = 96;  // 64
+    // walking / running sounds ------------------------
+    if (engine->config->settings.WalkSound.value()) {
+        pParty->walk_sound_timer -= pEventTimer->uTimeElapsed;
+
+        if (pParty->walk_sound_timer <= 0) {
+            pAudioPlayer->stopWalkingSounds();
+        }
+
+        // Start sound processing only when actual movement is performed to avoid
+        // stopping sounds on high FPS
+        if (pEventTimer->uTimeElapsed) {
```

Comment:
Does this actually happen? Are we already in the 1000+fps territory?

Reply:
Well, timer is defined in terms of 1/128th of second so 129 FPS is enough to always have 0 uTimeElapsed.

Reply:
I average ~1700fps walking round emerald